src/mlpack/core/data/serialization_shim.hpp

```diff
@@ -35,37 +35,49 @@ struct HasSerializeFunction
 {
   static const bool value =
       // Non-static version.
-      HasSerializeCheck<T, void(T::*)(boost::archive::xml_oarchive&,
-                                      const unsigned int)>::value ||
+      HasSerializeCheck<T,
+                        void (T::*)(boost::archive::xml_oarchive&,
+                                    const unsigned int)>::value ||
       // Static version.
-      HasSerializeCheck<T, void(*)(boost::archive::xml_oarchive&,
-                                   const unsigned int)>::value;
+      HasSerializeCheck<T,
+                        void (*)(boost::archive::xml_oarchive&,
+                                 const unsigned int)>::value;
 };
 
 template<typename T>
 struct HasSerialize
 {
   // We have to handle the case where T isn't a class...
   typedef char yes[1];
-  typedef char no [2];
-  template<typename U, typename V, typename W> struct check;
-  template<typename U> static yes& chk( // This matches classes.
+  typedef char no[2];
+  template<typename U, typename V, typename W>
+  struct check;
+  template<typename U>
+  static yes& chk( // This matches classes.
       check<U,
             typename std::enable_if_t<std::is_class<U>::value>*,
             typename std::enable_if_t<HasSerializeFunction<U>::value>*>*);
-  template<typename  > static no&  chk(...); // This matches non-classes.
+  template<typename>
+  static no& chk(...); // This matches non-classes.
 
   static const bool value = (sizeof(chk<T>(0)) == sizeof(yes));
 };
 
 // Declare the shims we need.
-template<typename T> struct FirstShim;
-template<typename T> struct FirstArrayShim;
-template<typename T> struct FirstNormalArrayShim;
-template<typename T> struct SecondShim;
-template<typename T> struct SecondArrayShim;
-template<typename T> struct SecondNormalArrayShim;
-template<typename T> struct PointerShim;
+template<typename T>
+struct FirstShim;
+template<typename T>
+struct FirstArrayShim;
+template<typename T>
+struct FirstNormalArrayShim;
+template<typename T>
+struct SecondShim;
+template<typename T>
+struct SecondArrayShim;
+template<typename T>
+struct SecondNormalArrayShim;
+template<typename T>
+struct PointerShim;
```

Comment:
There is an option `AlwaysBreakTemplateDeclarations` to handle this. It has two options `true` or `false`. 

```c++
true:                                  false:
template <typename T>          vs.     template <typename T> class C {};
class C {};
```
In most of the mlpack codebase, I have observed that there is always break after template declaration statements. So I set it to `true`, hence this change is there. If I change it to `false`, all the template declaration statements will change. What should we do?